selfdrive/locationd/paramsd.cc

```diff
@@ -105,17 +106,12 @@ int main(int argc, char *argv[]) {
   int save_counter = 0;
   while (true){
     for (auto s : poller->poll(100)){
-      Message * msg = s->receive();
+      MessageReader amsg = s->receive();
+      if (!amsg) continue;
 
-      auto amsg = kj::heapArray<capnp::word>((msg->getSize() / sizeof(capnp::word)) + 1);
-      memcpy(amsg.begin(), msg->getData(), msg->getSize());
+      localizer.handle_log(amsg.getEvent());
```

Comment:
Two calls to getEvent here, is msg_reader->getRoot<cereal::Event>(); fast?

Reply:
msg_reader->getRoot<cereal::Event>()  is called only once when msg_reader is not initialized,no matter how many times the getEvent() function is invoked. 
getEvent()  only return a reference to  member MessageRead.event. it's extreme fast.